lightningd/plugin.c

```diff
@@ -1086,9 +1087,35 @@ void json_add_opt_plugins(struct json_stream *response,
 			  const struct plugins *plugins)
 {
 	struct plugin *p;
+	struct plugin_opt *opt;
+	const char *opt_name;
+	const char *plugin_name;
+	char *plugin_ext;
+
+	/* we output 'plugins' and their config as substructure:
+	 * "plugins" : { "name" -> { "param" : "value" }, ... }  */
+	json_object_start(response, "plugins");
 	list_for_each(&plugins->plugins, p, list) {
-		json_add_string(response, "plugin", p->cmd);
+
+		/* guess a suitable plugin name: basename without extension */
+		plugin_name = tal_strdup(p, basename(p->cmd));
+		plugin_ext = strrchr(plugin_name, '.');
+		if (plugin_ext != NULL)
+			plugin_ext[0] = 0;
+		json_object_start(response, plugin_name);
+		tal_free(plugin_name);
+
+		json_add_string(response, "_cmd", p->cmd);
+		list_for_each(&p->plugin_opts, opt, list) {
+			/* Trim the `--` that we added before */
+			opt_name = opt->name + 2;
+			if (opt->value->as_str) {
```

Comment:
we should probably still include it if it's not a string? or at least include the option name?

Reply:
or at least add a FIXME to change this when/if we allow non-string values for plugin options.

Reply:
Okay, bool and ints are now also returned. Anything other than string, bool and int will be mapped to a `null` so we have at least the key.

Reply:
```
        "options": {
            "autocleaninvoice-cycle": null,
            "autocleaninvoice-expired-by": null
         }
```